### PR TITLE
Loop Mutations

### PR DESCRIPTION
**Pull request by cynthisl**
Multiple mutations per operator - return the mutations as a dict <name, mutation> (key name doesn't matter too much right now). The type is check and if it's a dict, split it up and handle those multiples.

Adding loop mutations:
* For loop
  * Run loop body zero times - replace iteration list with empty
  * Run loop body exactly once - add break to end of loop body
* While loop
  * Run loop zero times - add break immediately after while statement
  * Run loop exactly once - add break to the end of loop body

Still in work:
* List comprehension For
  * Mutation done, but unit test still needed

### CODE REVIEW

File: mutmut/__init__.py [hunk 1]
```diff
@@ -417,13 +417,16 @@ def loop_mutation(children, node, **_):
     mutations = {}
     # need to deep copy inside subfuctions, otherwise both mutations applied at same time
```
Comment (kc7zep) [inline mutmut/__init__.py]: typo: subfuctions => subfunctions

File: mutmut/__init__.py [hunk 2]
```diff
@@ -637,7 +665,7 @@ def mutate_node(node, context):
             # I guess a set might be fine too, best would be a custom data struct
             new_mutations = []
             if type(new_evaluation) == dict:
-                print("Multiple mutations for %s", node.get_code())
+                #print("Multiple mutations for %s", node.get_code())
```
Comment (kc7zep) [inline mutmut/__init__.py]: Remove this line?

File: mutmut/__init__.py [hunk 3]
```diff
@@ -405,28 +407,78 @@ def loop_mutation(children, node, **_):
     """
     Mutates loop
 
-    Zero-run loop: Replaces iteration list with a blank list []
-        Surviving Test Case: Only write test case that checks for an empty loop
-        
+    For loop children is structered as the nodes that make up the loop defintion (for x in y:) and a suite
```
Comment (kc7zep) [inline mutmut/__init__.py]: typo: structered => structured

File: mutmut/__init__.py [hunk 4]
```diff
@@ -405,28 +407,78 @@ def loop_mutation(children, node, **_):
...
+    The suite is a newline, indented, statement, then dedent and naother newline
```
Comment (kc7zep) [inline mutmut/__init__.py]: typo: naother => another

File: mutmut/__init__.py [hunk 5]
```diff
@@ -518,7 +518,6 @@ def list_comprehension_mutation(children, node, **_):
     # find in, everything asfter that is the list
     children = children[:]
     list_idx = None
-    pdb.set_trace()
```
Comment (kc7zep) [inline mutmut/__init__.py]: Can you also remove the
import pdb
at the top of this file?

File: mutmut/__init__.py (hunk 1 above)
Comment (cynthisl) [reply mutmut/__init__.py]: Fixed

File: mutmut/__init__.py (hunk 2 above)
Comment (cynthisl) [reply mutmut/__init__.py]: Done

File: mutmut/__init__.py (hunk 3 above)
Comment (cynthisl) [reply mutmut/__init__.py]: Fixed

File: mutmut/__init__.py (hunk 4 above)
Comment (cynthisl) [reply mutmut/__init__.py]: Fixed

File: mutmut/__init__.py (hunk 5 above)
Comment (cynthisl) [reply mutmut/__init__.py]: Whoops, deleted

**Pull request merged**